vrrp/vrrpv3-topo-1-1/munet.yaml

```diff
@@ -39,30 +40,34 @@ kinds:
           local snapshot_dir=$1
           local sleep_duration=$2
           local events_file=$3
+          local toponame=$4
+
           nodename=${MUNET_NODENAME}
 
           # Wait for initial network convergence.
           sleep "${sleep_duration}"
 
           # Ensure snapshot directory exists.
-          mkdir -p "${snapshot_dir}/${nodename}"
+          mkdir -p "${snapshot_dir}/${toponame}/${nodename}"
 
           # Collect running configuration in JSON format.
-          holo-cli -c "show running format json" > "${snapshot_dir}/${nodename}/config.json"
+          holo-cli -c "show running format json" > "${snapshot_dir}/${toponame}/${nodename}/config.json"
 
           # Collect protocol events.
-          cp "/tmp/munet/${nodename}/var.run.holo/${events_file}" "${snapshot_dir}/${nodename}/events.jsonl"
-          sed -i '/"Northbound":/d' "${snapshot_dir}/${nodename}/events.jsonl"
+          cp "/tmp/munet/${nodename}/var.run.holo/${events_file}" "${snapshot_dir}/${toponame}/${nodename}/events.jsonl"
+          sed -i '/"Northbound":/d' "${snapshot_dir}/${toponame}/${nodename}/events.jsonl"
       }
 
       if [[ -n "${SNAPSHOT_DIR}" ]]; then
-          collect_network_snapshot "${SNAPSHOT_DIR}" 10 "holo-events-vrrp-eth-sw1.jsonl"
+          export SNAPSHOT_DIR=${SNAPSHOT_DIR}/holo-vrrp/tests/conformance/topologies/
+          collect_network_snapshot "${SNAPSHOT_DIR}" 10 "holo-events-vrrp-eth-sw1.jsonl" "vrrpv3-topo1-1"
       fi
 
     volumes:
       - "./%NAME%/etc.holo:/etc/holo/"
       - "%RUNDIR%/var.log:/var/log"
       - "%RUNDIR%/var.run.holo:/var/run/holo"
+      - "%CONFIGDIR%/../env.txt:/etc/holo-env.txt"
```

Comment:
hmm this is very interesting. Couldn't we make that a shell script like `holo-setup.sh` and move everything there? Right now we have to duplicate the `collect_network_snapshot` function in all topologies and that's really bad from a maintenance perspective.

Reply:
Done...Can probably do better aesthetically on the script.

But it works. Let me know what you think. If you're okay with it, I can get this running on all vrrp topos then we can progressively do the same for the rest.

Reply:
Thanks for doing this, the `setup/setup.sh` file is exactly what we need. Please mark this as ready for review and I'll merge it!